modules/user_accounts/php/NDB_Form_user_accounts.class.inc

```diff
@@ -1,48 +1,82 @@
 <?php
+/**
+* The user account page
+*
+* PHP Version 5
+*
+* @category Main
+* @package  User_Account
+* @author   Loris Team <loris.mni@bic.mni.mcgill.ca>
+* @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+* @link     https://www.github.com/aces/Loris/
+*/
+
 require_once "NDB_Form.class.inc";
 require_once "Email.class.inc";
 
 /**
- * The forms for the user accounts menu
- * @package main
- */
-class NDB_Form_user_accounts extends NDB_Form
+* Implements the user account page
+*
+* @category Main
+* @package  User_Account
+* @author   Loris Team <loris.mni@bic.mni.mcgill.ca>
+* @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+* @link     https://www.github.com/aces/Loris/
+*/
+class NDB_Form_User_Accounts extends NDB_Form
 {
+    /**
+     * Controls who's got access to this page, namely those who have the
+     * 'user_accounts' and who either have permission
+     * 'user_accounts_multisite' or whose site matches the site of the user they
+     * are trying to edit.
+     *
+     * @return true if user has access, false otherwise.
+     */
     function _hasAccess()
     {
         // create user object
         $editor =& User::singleton();
-        if (Utility::isErrorX($editor)) {
-            return PEAR::raiseError("User Error: ".$editor->getMessage());
-        }
 
         if ($this->page == 'edit_user') {
-            if(!empty($this->identifier)) {
+            if (!empty($this->identifier)) {
                 $user =& User::factory($this->identifier);
-                if(Utility::isErrorX($user)) {
-                    return PEAR::raiseError("User Error ($this->identifier): ".$user->getMessage());
+            }
+
+            if ($editor->hasPermission('user_accounts')) {
+                if ($editor->hasPermission('user_accounts_multisite')) {
+                    return true;
                 }
+
+                if (empty($this->identifier)) {
+                    return true;
+                }
+
+                return $editor->getData('CenterID') == $user->getData('CenterID');
             }
```

Comment:
The original return statement, before my pull request was like this:

            return ($editor->hasPermission('user_accounts') &&  ($editor->hasPermission('user_accounts_multisite') || (empty($this->identifier) || $editor->getData('CenterID') == $user->getData('CenterID'))));

I just split it into multiple returns and I think it boils down to the exact same logic. If you do not think so, please provide counter-example.